DomainServices/Repositories/CustomersRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using DomainModels.ExtensionMethods;
+using DomainModels.Models;
+using EntityFrameworkCore.Repository.Interfaces;
+using EntityFrameworkCore.UnitOfWork.Interfaces;
+using Infrastructure.Data.Context;
+
+namespace DomainServices.Repositories
+{
+    public class CustomerRepository : ICustomerRepository
+    {
+        private readonly IRepositoryFactory _repositoryFactory;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IRepository<Customer> _customerRepository;
+
+        public CustomerRepository(IUnitOfWork<WarrenEverestContext> unitOfWork, IRepositoryFactory<WarrenEverestContext> repository)
+        {
+            _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+            _repositoryFactory = repository ?? (IRepositoryFactory)_unitOfWork;
+            _customerRepository = _repositoryFactory.Repository<Customer>();
+        }
```

Comment:
Não fazemos isso nessa lib, não é a mesma coisa que o DbSet
```suggestion
        }
```